linera-chain/src/chain.rs

```diff
@@ -239,10 +239,16 @@ pub struct ChainStateView<C>
 where
     C: Clone + Context + Send + Sync + 'static,
 {
-    /// Execution state, including system and user applications.
-    pub execution_state: ExecutionStateView<C>,
     /// Hash of the execution state.
     pub execution_state_hash: RegisterView<C, Option<CryptoHash>>,
+    /// Execution state, including system and user applications.
+    /// NOTE: This has to **not** be the first field in `ChainStateView` in order for execution
+    /// state caching to work. This is because the derivation of the `RootView` trait delegates
+    /// `context()` calls to the first field. If we cache the field, we also save the storage LRU
+    /// cache (if enabled), which is a part of the context, and restoring its value from the cache
+    /// results in storage using an outdated cache, which may result in incorrect values being
+    /// returned by read operations.
```

Comment:
I think it would be better if this didn't show up in GraphQL. Also, it doesn't affect any user of this type; it's only relevant if one wants to modify the type.

It would also be good to have a ticket for a proper solution, and link to it with a `TODO` here.

```suggestion
    // NOTE: This has to **not** be the first field in `ChainStateView` in order for execution
    // state caching to work. This is because the derivation of the `RootView` trait delegates
    // `context()` calls to the first field. If we cache the field, we also save the storage LRU
    // cache (if enabled), which is a part of the context, and restoring its value from the cache
    // results in storage using an outdated cache, which may result in incorrect values being
    // returned by read operations.
```